app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js

```diff
@@ -1,33 +1,52 @@
-import React from 'react';
-
-const ExpandedStoryControls = ({ onCancel, onSave, onDelete, readOnly }) => {
-  return (
-    <div className="form-group Story__controls">
-      <input className="save"
-        onClick={onSave}
-        type="button"
-        value={I18n.t('save')}
-        disabled={readOnly}
-      />
-
-      <input className="delete"
-        onClick={() => {
-          if (window.confirm(I18n.t('story destroy confirm'))){
-            onDelete();
-          }
-        }}
-        type="button"
-        value={I18n.t('delete')}
-        disabled={readOnly}
-      />
-
-      <input className="cancel"
-        onClick={onCancel}
-        type="button"
-        value={I18n.t('cancel')}
-      />
-    </div>
-  );
+import React, { Component } from 'react';
+
+class ExpandedStoryControls extends Component {
```

Comment:
Why did you change this functional component to a class?
It looks like you are not using internal state neither a lifecycle method.

Reply:
With the addition of more logic when canceling a story, I've created the method `handleCancel` and took the liberty of creating a class in order to encapsulate the logic into a single component.

Reply:
It was already a single component, you could just have used another function to `handleCancel`.
To put it simple, I just create class components when I need to use internal state or lifecycle methods, that's why I asked. But no problems there.

Reply:
I see, thanks for the heads up!

Reply:
@gutofoletto two things here:

1 - when reviewing, try to give context. For [your first comment ](https://github.com/Codeminer42/cm42-central/pull/508/files#r255657239) you make no parallel why internal state and lifecycle methods are related to functional components, you may commenting in code from people that didn't work with React before and PR reviews are meant to help people to learn.
2 - Avoid using "just" when suggesting changes, it may not be that simple for the person that is colaborating.

This being said, I disagree about creating two another functions (`handleDelete` and `handleCancel`) for that because functions give the impression they're pure when used for this kind of problem which it definitelly won't be because `window.confirm` isn't deterministic. Having a bunch of functions around won't make the code better than having a class that handles what we're doing in an encapsulated and easier to read way.